arwen/elrondapi/elrondei.go

```diff
@@ -300,10 +319,69 @@ func getOwner(context unsafe.Pointer, resultOffset int32) {
 		return
 	}
 
-	gasToUse := metering.GasSchedule().ElrondAPICost.GetOwner
+	gasToUse := metering.GasSchedule().ElrondAPICost.GetSCAddress
 	metering.UseGas(gasToUse)
 }
 
+//export getOwnerAddress
+func getOwnerAddress(context unsafe.Pointer, resultOffset int32) {
+	blockchain := arwen.GetBlockchainContext(context)
+	runtime := arwen.GetRuntimeContext(context)
+	metering := arwen.GetMeteringContext(context)
+
+	owner, err := blockchain.GetOwnerAddress()
+	if arwen.WithFault(err, context, runtime.ElrondAPIErrorShouldFailExecution()) {
+		return
+	}
+
+	err = runtime.MemStore(resultOffset, owner)
+	if arwen.WithFault(err, context, runtime.ElrondAPIErrorShouldFailExecution()) {
+		return
+	}
+
+	gasToUse := metering.GasSchedule().ElrondAPICost.GetOwnerAddress
+	metering.UseGas(gasToUse)
+}
+
+//export getShardOfAddress
+func getShardOfAddress(context unsafe.Pointer, addressOffset int32) int32 {
+	blockchain := arwen.GetBlockchainContext(context)
+	runtime := arwen.GetRuntimeContext(context)
+	metering := arwen.GetMeteringContext(context)
+
+	address, err := runtime.MemLoad(addressOffset, arwen.AddressLen)
+	if arwen.WithFault(err, context, runtime.ElrondAPIErrorShouldFailExecution()) {
+		return 0
+	}
+
+	gasToUse := metering.GasSchedule().ElrondAPICost.GetShardOfAddress
+	metering.UseGas(gasToUse)
+
+	return int32(blockchain.GetShardOfAddress(address))
+}
+
+//export isSmartContract
+func isSmartContract(context unsafe.Pointer, addressOffset int32) int32 {
+	blockchain := arwen.GetBlockchainContext(context)
+	runtime := arwen.GetRuntimeContext(context)
+	metering := arwen.GetMeteringContext(context)
+
+	address, err := runtime.MemLoad(addressOffset, arwen.AddressLen)
+	if arwen.WithFault(err, context, runtime.ElrondAPIErrorShouldFailExecution()) {
+		return 0
+	}
+
+	gasToUse := metering.GasSchedule().ElrondAPICost.IsSmartContract
+	metering.UseGas(gasToUse)
+
+	isSmartContract := blockchain.IsSmartContract(address)
+	if isSmartContract {
+		return 1
```

Comment:
Maybe we should have constants for TRUE and FALSE?

Reply:
Fixed by adding a helper function, `BooleanToInt`. Is this ok?